src/shared/btrfs-util.c

```diff
@@ -1637,13 +1637,10 @@ int btrfs_qgroup_find_parents(int fd, uint64_t qgroupid, uint64_t **ret) {
                         break;
         }
 
-        if (n_items <= 0) {
-                *ret = NULL;
-                return 0;
-        }
+        assert((n_items > 0) == (items != NULL));
+        assert(n_items <= INT_MAX);
```

Comment:
Is https://github.com/systemd/systemd/commit/5e30e6e28190fe30e76e071b2eb99546abcee7e6 no longer necessary with this?

Reply:
Possibly, but that's just a different check in a different place. I saw Luca's patch, and I think we can have both.